File: Search/In C/BinarySearch.c
#include<stdio.h>
#include<conio.h>

void getArray(int array[],int n){

    for(int i=0;i<n;i++){
        scanf("%d",&array[i]);
    }
}

void dispArray(int array[],int n){

    for(int i=0;i<n;i++){
        printf("%d ",array[i]);
    }   
}

int BinarySearch(int array[],int key,int low,int high){
    int mid=(low+high)/2;
    while(mid){
        if(array[mid]==key)
            return mid;
        else if(key<array[mid])
            return BinarySearch(array,key,low,mid-1);
        else    
            return BinarySearch(array,key,mid+1,high);
    }
    return -1;
    }

int main(){

    int n,array[50],key,r;
    printf("Enter number of elements in the array: ");
    scanf("%d",&n);
    getArray(array,n);

    printf("Enter element to search: ");
    scanf("%d",&key,n);

    
    printf("The given array is ");
    dispArray(array,n);
    r=BinarySearch(array,key,0,n-1);

    if(r!=-1)
        printf("\nThe element %d is found at index %d",key,r);
    else
        printf("\nThe element is not found");

}

File: Search/In Python #prefer/alternate/binary-search.py
# Binary Search in python

array=[]
n=int(input("Enter Number of elements in the array: "))

print("Enter elemens one by one: ")

for i in range(n):
    e=int(input())
    array.append(e)
    
x=int(input("Enter element to Search: "))

def binarySearch(array, x, low, high):

    # Repeat until the pointers low and high meet each other
    while low <= high:

        mid = low + (high - low)//2

        if array[mid] == x:
            return mid

        elif array[mid] < x:
            low = mid + 1

        else:
            high = mid - 1

    return -1


result = binarySearch(array, x, 0, len(array)-1)

if result != -1:
    print("Element is present at index " + str(result))
else:
    print("Not found")